### PR TITLE
Draft 24 updates

### LINKED ISSUE #483: Update to draft 24
- [x] Transport: #496 
- [x] TLS: ^
- [x] Recovery: ^

### PR DESCRIPTION
**Pull request by Ralith**
Now covering all the changes. Fixes #483.

### CODE REVIEW

File: quinn-proto/src/shared.rs [hunk 1]
```diff
@@ -209,9 +209,10 @@ impl TransportConfig {
 pub struct EndpointConfig {
     /// Length of connection IDs for the endpoint.
     ///
-    /// This must be either 0 or between 4 and 18 inclusive. The length of the local connection IDs
-    /// constrains the amount of simultaneous connections the endpoint can maintain. The API user is
-    /// responsible for making sure that the pool is large enough to cover the intended usage.
+    /// This must be no greater than 20. If zero, incoming packets are mapped to connections only by
+    /// their source address. Otherwise, the connection ID field is used alone, allowing for source
+    /// address to change and for multiple connections from a single address. Settings of 1 or 2 may
+    /// slow or prevent processing of large numbers of connections.
```
[the remark this answers is not in the document]
Comment (Ralith) [reply quinn-proto/src/shared.rs]: I'm not sure I'd call it that. If you have an 8-bit connection ID space, and you want to handle 257 connections in a single endpoint, there's just nothing you can do.

File: quinn-proto/src/endpoint.rs [hunk 2]
```diff
@@ -47,7 +47,7 @@ where
     ///
     /// Incoming stateless resets do not have correct CIDs, so we need this to identify the correct
     /// recipient, if any.
-    connection_reset_tokens: HashMap<ResetToken, ConnectionHandle>,
+    connection_reset_tokens: HashMap<SocketAddr, HashMap<ResetToken, ConnectionHandle>>,
```
[the remark this answers is not in the document]
Comment (Ralith) [reply quinn-proto/src/endpoint.rs]: Done.

File: quinn-proto/src/shared.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (djc) [reply quinn-proto/src/shared.rs]: I'm just wondering how Quinn actually behaves if it gets the 257th connection on an 8-bit connection ID endpoint (actually, birthday paradox applies so the problem would happen much sooner, of course). Would it get slower, prevent processing of further connections, or both?

[the remark this answers is not in the document]
Comment (Ralith) [reply quinn-proto/src/shared.rs]: `new_cid` will progressively get slower, until it ultimately can no longer terminate. Come to think of it, we should probably just return `SERVER_BUSY` if more than, say, 3/4 of CIDs are allocated...

[the remark this answers is not in the document]
Comment (Ralith) [reply quinn-proto/src/shared.rs]: See #510.

**Comment by djc** [comment]
Good stuff! I did find the `next_crypto` stuff a bit ugly looking, what with all those `unwrap()`s that are hard to statically check. I wonder if we could come up with a different design for this.

**Pull request merged**